src/components/higlass/HiGlassSubscriber.js

```diff
@@ -111,16 +123,98 @@ export default function HiGlassSubscriber(props) {
-      options={{
-        ...hgOptions,
-        theme,
-      }}
-    />
-  ), [hgViewConfig, hgOptions, theme]);
+  const hgOptions = useMemo(() => ({
+    ...hgOptionsProp,
+    theme,
+  }), [hgOptionsProp, theme]);
+
+  const hgViewConfig = useMemo(() => {
+    // HiGlass needs the start and end absolute genome coordinates
+    const initialXDomain = [
+      genomicTargetX - ((genomeSize / (2 ** genomicZoomX))) * ((width / 2) / HG_SIZE),
+      genomicTargetX + ((genomeSize / (2 ** genomicZoomX))) * ((width / 2) / HG_SIZE),
+    ];
+    const initialYDomain = [
+      genomicTargetY - ((genomeSize / (2 ** genomicZoomY))) * ((height / 2) / HG_SIZE),
+      genomicTargetY + ((genomeSize / (2 ** genomicZoomY))) * ((height / 2) / HG_SIZE),
+    ];
+    return {
+      editable: false,
+      zoomFixed: false,
+      trackSourceServers: [
+        '//higlass.io/api/v1',
+      ],
+      exportViewUrl: '//higlass.io/api/v1/viewconfs',
+      views: [
+        {
+          ...hgViewConfigProp,
+          initialXDomain,
+          initialYDomain,
+        },
+      ],
+      zoomLocks: {
+        locksByViewUid: {},
+        locksDict: {},
+      },
+      locationLocks: {
+        locksByViewUid: {},
+        locksDict: {},
+      },
+      valueScaleLocks: {
+        locksByViewUid: {},
+        locksDict: {},
+      },
+    };
+  }, [genomicTargetX, genomeSize, genomicZoomX, width, genomicTargetY,
+    genomicZoomY, height, hgViewConfigProp]);
+
+  useEffect(() => {
+    const handleMouseEnter = () => {
+      isActiveRef.current = true;
+    };
+    const handleMouseLeave = () => {
+      isActiveRef.current = false;
+    };
+    const container = containerRef.current;
+    container.addEventListener('mouseenter', handleMouseEnter);
+    container.addEventListener('mouseleave', handleMouseLeave);
+    return () => {
+      container.removeEventListener('mouseenter', handleMouseEnter);
+      container.removeEventListener('mouseenter', handleMouseLeave);
+    };
+  }, [containerRef]);
+
+
+  useEffect(() => {
+    if (!hgInstance) {
+      return () => {};
+    }
+    hgInstance.api.on('viewConfig', (viewConfigString) => {
+      const viewConfig = JSON.parse(viewConfigString);
+      const xDomain = viewConfig.views[0].initialXDomain;
+      const yDomain = viewConfig.views[0].initialYDomain;
+
+      const nextGenomicZoomX = Math.log2(
+        genomeSize / ((xDomain[1] - xDomain[0]) * (HG_SIZE / width)),
+      );
+      const nextGenomicZoomY = Math.log2(
+        genomeSize / ((yDomain[1] - yDomain[0]) * (HG_SIZE / height)),
+      );
+      const nextGenomicTargetX = (xDomain[0] + (xDomain[1] - xDomain[0]) / 2);
+      const nextGenomicTargetY = (yDomain[0] + (yDomain[1] - yDomain[0]) / 2);
+      // Only set if the user mouse is over this component ("is active").
+      // Otherwise, this could be an initial on viewConfig change callback from a sibling,
+      // which will cause an infinite loop.
+      if (isActiveRef.current) {
```

Comment:
Can we just check this whole function for this condition?

Reply:
Yes